examples/compute/main.rs

```diff
@@ -0,0 +1,305 @@
+                            ],
+                        )],
+                        options: PipelineOptions::default(),
+                    },
+                );
+            }
+        }
+
+        // Run the pipeline
+        let work_group_size = spawner.work_group_size;
+        renderer.compute(
+            &mut compute.pipeline,
+            WorkGroups {
+                x: work_group_size,
+                y: 1,
+                z: 1,
+            },
+        );
+    }
+}
+
+fn render(
+    globals: Const<Globals>,
+    world: Const<World>,
+    assets: Const<Assets>,
+    mut renderer: Mut<Renderer>,
+) {
+    for (spawner, pipeline) in world.query::<(&mut ParticlesSpawner, &mut Pipeline)>() {
+        // Set the shader on the pipeline
+        if pipeline.shader.is_null() {
+            pipeline.shader = assets.find::<Shader>(RENDER_PIPELINE).unwrap_or_default();
+        }
+
+        let mesh = assets
+            .get(assets.find(PARTICLE_MESH).expect("Mesh must be loaded"))
+            .unwrap();
+
+        if !pipeline.ready() {
+            if let Some(shader) = assets.get(pipeline.shader) {
+                if !shader.loaded() {
+                    continue;
+                }
+
+                // Get the uniform that holds the camera's project view
+                let proj_view = globals
+                    .get::<ProjView>()
+                    .expect("ProjView buffer must be loaded");
+
+                renderer.bind(
+                    pipeline,
+                    PipelineLayout {
+                        label: String::from(RENDER_PIPELINE),
+                        mesh: Some(mesh),
+                        shader,
+                        bindings: &[BindGroup::new(
+                            "Globals",
+                            vec![
+                                Binding::Uniform("ProjView", Stage::Vertex, &proj_view.uniform),
+                                Binding::StorageAsUniform(
+                                    "Particles",
+                                    Stage::Vertex,
+                                    &spawner.particles,
+                                ),
+                            ],
+                        )],
+                        options: PipelineOptions::default(),
+                    },
+                );
+            }
+        }
+
+        renderer.run(pipeline, mesh);
+    }
+}
+
+fn ui(overlay: Const<Overlay>, frame: Const<Frame>) {
+    let egui_overlay = overlay
+        .get::<Egui>()
+        .expect("Egui overlay must be added on startup");
+
+    egui::Area::new("FPS counter")
+        .fixed_pos(egui::pos2(16.0, 16.0))
+        .show(&egui_overlay.ctx, |ui| {
+            ui.colored_label(
+                egui::Rgba::from_rgb(255.0, 255.0, 255.0),
+                format!("FPS: {:.1}", frame.fps()),
+            );
+        });
+}
+
+fn cube(width: f32) -> Mesh {
```

Comment:
Cube is also used in shader example and it's a common primitive. We can add some primitives into dotrix itself and use it for example as
```
Mesh::cube(size)
```

Reply:
I've introduced a new crate called `dotrix_primitives` and updated the examples. The crate will contain builders for meshes.